### ISSUE TITLE
Concept of multiple private keys per user

### ISSUE THREAD

**Issue by MaxFichtelmann** [nextcloud/end_to_end_encryption_rfc#30]
The current assumption is that every user has a single private key.

Cryptography based on elliptic curves often has better properties regarding performance and size, so it may be desirable for a user to use that instead of RSA. I do not think there is an immediate need to implement EC based public key crypto here, but the assumption of a single private key per use may hurt future adoption quite a lot.

There may be other reasons to maintain multiple private keys.

TLDR; please consider revising the assumption that a user has a single private key.
